MyLinkedList.java

```diff
@@ -1 +1,147 @@
+package com.company;
 
+public class MyLinkedList<T> implements List<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private StringBuilder stringBuilder = new StringBuilder();
+
+    @Override
+    public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            add(value);
+        }
+        if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        }
+        Node<T> leftNode = currentNode(index);
+        Node<T> rightNode = leftNode.prev;
+        leftNode.next = newNode;
```

Comment:
maybe leftNode.prev = newNode;
and change other links?

Reply:
Sorry, i don't understand what do you mean

Reply:
does this method work correctly?
if we have list: 1, 2, 4, 5.
and write: add(3, 2);
we should have the result: 1, 2, 3, 4, 5

Reply:
yes. i corrected it